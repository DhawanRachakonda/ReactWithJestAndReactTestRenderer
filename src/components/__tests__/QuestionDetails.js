import React from 'react';
import renderer from 'react-test-renderer';
import { mapStateToProps, QuestionDetailDisplay } from '../QuestionDetail.jsx';

describe('The Question Detail Component', () => {
    describe('The container element', ()=>{
        describe('mapStateToProps', ()=>{
            it('should map the state to props correctely', ()=>{
                const sampleQuestion = {
                    question_id: 42,
                    body: 'What is PureComponent',
                };
                const appState = {
                    questions: [ sampleQuestion ],
                };
                const ownProps = {
                    question_id: 42,
                };
                const componentState = mapStateToProps(appState, ownProps);
                expect(componentState).toEqual(sampleQuestion);
            });
        });
    });
    it('The display element', () => {
        const tree = renderer.create(
            <QuestionDetailDisplay
                title="Hooks"
                body="What are hooks in React"
                answer_count={2}
                tags={['React']}
            />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
});