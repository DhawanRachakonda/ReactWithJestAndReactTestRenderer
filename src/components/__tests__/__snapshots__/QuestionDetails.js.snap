// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Question Detail Component The display element 1`] = `
<div>
  <h3
    className="mb-2"
  >
    Hooks
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          React
        </code>
      </div>
    </div>
    <div>
      <p>
        What are hooks in React
      </p>
    </div>
    <div>
      2
       Answers
    </div>
  </div>
</div>
`;
